src/platform/activity/activity.service.ts

```diff
@@ -118,12 +123,26 @@ export class ActivityService {
       visibility: visibility,
     });
 
+    if (userID) {
+      qb.andWhere({ triggeredBy: userID });
+    }
+
+    if (excludeTypes && excludeTypes.length > 0) {
+      qb.andWhere({
+        type: Not(In(excludeTypes)),
+      });
+    }
+
     if (types && types.length > 0) {
-      qb.andWhere({ type: In(types) });
+      const filteredTypes =
+        excludeTypes && excludeTypes.length > 0
+          ? types.filter(type => !excludeTypes.includes(type))
+          : types;
+      qb.andWhere({ type: In(filteredTypes) });
```

Comment:
I would just leave the previous statement - it's not generating an ideal SQL statement, but it's so much simpler code wise.